src/utils/Utils.ts

```diff
@@ -1341,6 +1341,32 @@ export default class Utils {
     return /^[A-Z0-9- ]*$/.test(plateID);
   }
 
+  public static separateCryptoAlgo(algo: string): CryptoKeyProperties {
+    const regExp = /(.*?)-(.*?)-(.*)/.exec(algo);
```

Comment:
Just split the string with `algo.split('-')`. With object destructuring to array you can have in one line: 

`[blockCypher, blockSize, operationMode] = algo.split('-')`

Regexp is overkill here and have performance drawbacks than simple string handling methods do not exhibit.